lib/rules/max-len.js

```diff
@@ -157,11 +157,14 @@ module.exports = {
          */
         function isFullLineComment(line, lineNumber, comment) {
             var start = comment.loc.start,
-                end = comment.loc.end;
+                end = comment.loc.end,
+                isFirstTokenOnLine = !line.slice(0, comment.loc.start.column).trim(),
```

Comment:
That's a pretty arcane bit of code- would it be worth extracting that to a place like ast-utils?